tools/tools.go

```diff
@@ -0,0 +1,26 @@
+// +build tools
+
+// To install the following tools at the version used by this repo run:
+// $ make tools
+// or
+// $ go generate -tags tools tools/tools.go
```

Comment:
Does tools.go modify `go.mod` so we can pin versions? Or does that not work? 

If so, it seems like we should have `make tools` just call the `go generate` rather than have two separate implementation.

Reply:
I believe it will if you run this command directly, but `make tools` turns off `GO111MODULE` before running `go generate` on this file so it doesn't end up modifying `go.mod` when you run the make target. But if you just run the command above I believe it will modify `go.mod`.

Reply:
Maybe that's a nicer solution here? We could have two options that enable the module env var:

```
brian@localghost:waypoint % go generate -tags tools tools/tools.go                                                                                                                                                   ±[●][add-tools-bootstrap]
go: finding module for package github.com/vektra/mockery
go: finding module for package github.com/hashicorp/go-bindata
go: finding module for package github.com/mitchellh/protoc-gen-go-json
go: found github.com/hashicorp/go-bindata in github.com/hashicorp/go-bindata v3.0.7+incompatible
go: found github.com/mitchellh/protoc-gen-go-json in github.com/mitchellh/protoc-gen-go-json v0.0.0-20200917194518-364b693410ae
go: found github.com/vektra/mockery in github.com/vektra/mockery v1.1.2
```

Reply:
If this is just running `go install` we don't even need to import them, do we? I suppose it won't include them in the waypoint binary either way. I see the code switches off modules too, so that would mean that the imports here wouldn't honor a go.sum anyway...

Reply:
Yeah.... initially I had turned of `GO11MODULE` since that looks to be what [vault is doing](https://github.com/hashicorp/vault/blob/f0849708a59e02025ac9808e6c1d451553af048f/Makefile#L125-L134) but I notice now they're only doing it for the `go get` steps. I'll go ahead and remove it, I think that should solve this problem anyway!

Reply:
Also with that off, we probably don't need the `go get` part of this either?

Reply:
Hm, so. if you run go get with modules off, it will just download them into $GOPATH and ignore go.mod. But then when `go install` is run via `go generate`, it will use go.mod.

So I think you can skip the whole `get` phase entirely and just do `go generate` since they're referenced from `go.mod`.Does that work?

Reply:
Yeah, I'll remove the get phase completely. `go generate` works just fine! I'll update the PR to just do that.